src/shared/clients/iotClient.ts

```diff
@@ -0,0 +1,702 @@
+}
+
+export interface ListThingsResponse {
+    readonly things: IotThing[]
+    readonly nextToken: string | undefined
+}
+
+export interface UpdateThingRequest {
+    readonly thingName: Iot.ThingName
+}
+
+export interface CreateThingResponse {
+    readonly thing: IotThing
+}
+
+export interface ListCertificatesRequest {
+    readonly pageSize?: Iot.PageSize
+    readonly marker?: Iot.Marker
+    readonly ascendingOrder?: Iot.AscendingOrder
+}
+
+export interface ListCertificatesResponse {
+    readonly certificates: IotCertificate[]
+    readonly nextMarker: string | undefined
+}
+
+export interface ListThingCertificatesRequest {
+    readonly thingName: Iot.ThingName
+    readonly nextToken?: Iot.NextToken
+    readonly maxResults?: Iot.MaxResults
+}
+
+export interface ListThingCertificatesResponse {
+    readonly certificates: IotCertificate[]
+    readonly nextToken: string | undefined
+}
+
+export interface UpdateCertificateRequest {
+    readonly certificateId: Iot.CertificateId
+    readonly newStatus: Iot.CertificateStatus
+}
+
+export interface DeleteCertificateRequest {
+    readonly certificateId: Iot.CertificateId
+    readonly forceDelete: Iot.ForceDelete
+}
+
+export interface AttachThingPrincipalRequest {
+    readonly thingName: Iot.ThingName
+    readonly principal: Iot.Principal
+}
+
+export interface ListPoliciesRequest {
+    readonly principal?: Iot.Principal
+    readonly pageSize?: Iot.PageSize
+    readonly marker?: Iot.Marker
+    readonly ascendingOrder?: Iot.AscendingOrder
+}
+
+export interface ListPoliciesResponse {
+    readonly policies: IotPolicy[]
+    readonly nextMarker: string | undefined
+}
+
+export interface DeletePolicyRequest {
+    readonly policyName: Iot.PolicyName
+}
+
+export interface AttachPolicyRequest {
+    readonly policyName: Iot.PolicyName
+    readonly target: Iot.Target
+}
+
+export class DefaultIotClient {
+    public constructor(
+        private readonly regionCode: string,
+        private readonly iotProvider: (regionCode: string) => Promise<Iot> = createSdkClient
+    ) {}
+
+    private async createIot(): Promise<Iot> {
+        return this.iotProvider(this.regionCode)
+    }
+
+    /**
+     * Lists all things owned by the client.
+     *
+     *
+     * @throws Error if there is an error calling S3.
+     */
+    public async listAllThings(): Promise<Iot.ThingAttribute[]> {
+        const iot = await this.createIot()
+
+        let iotThings: Iot.ThingAttribute[]
+        try {
+            const output = await iot.listThings().promise()
+            iotThings = output.things ?? []
+        } catch (e) {
+            getLogger().error('Failed to list things: %O', e)
+            throw e
+        }
+        return iotThings
+    }
+
+    /**
+     * Lists Things owned by the client.
+     *
+     * @throws Error if there is an error calling IoT.
+     */
+    public async listThings(request?: ListThingsRequest): Promise<ListThingsResponse> {
```

Comment:
Take a look at some of our other client wrapper `list` operations, such as `listFunctions` in the Lambda client. We have a pattern for handling paginated APIs that you should be able to follow here and I'm hopeful will help clean this up.